lib/src/signed_video_h26x_auth.c

```diff
@@ -741,6 +741,7 @@ maybe_validate_gop(signed_video_t *self, h26x_nalu_t *nalu)
   latest->number_of_expected_picture_nalus = -1;
   latest->number_of_received_picture_nalus = -1;
   latest->number_of_pending_picture_nalus = -1;
+  latest->public_key_has_changed = false;
```

Comment:
If |public_key_has_changed| switch to true when decoding the SEI validation should fail. I suggest adding a check on line 570. Something like this:

if (latest->public_key_has_changed) valid = SV_AUTH_RESULT_NOT_OK;